src/sagemaker/amazon/pca.py

```diff
@@ -31,7 +31,7 @@ class PCA(AmazonAlgorithmEstimatorBase):
     extra_components = hp(name='extra_components', validate=lambda x: x >= 0 and isinstance(x, int),
                           validation_message="Value must be an integer greater than or equal to 0")
 
-    def __init__(self, role, train_instance_count, train_instance_type, num_components,
+    def __init__(self, role, train_instance_count, train_instance_type, num_components, default_mini_batch_size,
```

Comment:
I'd like to try to avoid including new required arguments to function signatures. Why can't this have a default value?

Reply:
This is a required parameter that has no default. Instead of constructor let's enforce by fit() in new algorithms and I will leave default logic here to avoid breaking existing calls.